File: RiemannIntegral/Python/RiemannIntegral.py
'''
This script implements Riemann's Integral
a - left border of integration
b - right border of integration 
n - number of rectangles between borders
tup - tuple containg coefficients of polynomial
'''

import math

def function(tup, x):

    result = 0
    
    for i in range(len(tup)):
        result += tup[i] * math.pow(x, len(tup) - 1 - i)
 
    return result 

def riemannIntegral(a, b, n, tup):

    d = (b - a) / n
    k = 0
    P = 0
    while k != n:
        x = a + d * k + d / 2
        #P += d * abs(function(tup, x))
        P += d * function(tup, x)
        k += 1

    return P

tupOne = (1, 0, 0)
tupTwo = (1, -1, -13, 13, 36, -36)

[print("x^2: \t %f, \t\t x^5-x^4-13x^3+13x^2+36x-36: \t %f" % (riemannIntegral(1, 3, i, tupOne), riemannIntegral(1, 3, i, tupTwo))) for i in range(2, 100)]

File: RiemannIntegral/C/RiemannIntegral.c
#include <stdio.h>
#include <math.h>

double polynomial(double coeff[], double x, int size)
{
	double result = 0.0;

	for(int i = 0; i < size; i++)
		result += coeff[i] * pow(x, size - 1 - i);
	
	return result;
}

double riemannIntegral(double a, double b, double * n, double coeff[], int size)
{
	double d = (b - a) / *n, P = 0.0, x = 0.0;
	int k = 0;

	while(k != *n)
	{
		x = a + d * k + d / 2.0;
	      	P += d * polynomial(coeff, x, size);
		k++;
	}

	return P;
}

int main()
{
	double arrOne[] = {1.0, 0.0, 0.0};
	double arrTwo[] = {1.0, -1.0, -13.0, 13.0, 36.0, -36.0};
	
	for(double i = 2.0; i < 100.0; i++)
	{
		printf("f1: %.10f, \t\t f2: %.10f\n", riemannIntegral(1, 3, &i, arrOne, sizeof(arrOne)/sizeof(arrOne[0])), riemannIntegral(1, 3, &i, arrTwo, sizeof(arrTwo)/sizeof(arrTwo[0])));
	}
	return 0;
}
